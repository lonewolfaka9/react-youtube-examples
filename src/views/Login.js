import React, { Component } from 'react'
//import AppHeader from '../components/AppHeader';
import cookie from 'react-cookies'

import { observable } from 'mobx';
  
class loginStore {
  @observable username = "";
  @observable password ="";
}

  class Login extends Component {

  constructor(props) {
    super(props);

   
  

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

  }

  componentWillMount() {
    var authCookieValue = cookie.load("auth");// should be replaced by auth token
    if (authCookieValue === undefined) {
      authCookieValue = false;
    } else {
      authCookieValue = JSON.parse(authCookieValue);
    }
    let userAuth = authCookieValue;
    if (authCookieValue) { this.props.history.push("/search") };

  }

  handleChange(event) {
    const { id, value } = event.target;
    this[id] = value;
    console.log(this.username, value);

  }

  handleSubmit() {
    const { username, password } = this;
      // hard coded auth logic
    if (username === "admin" && password === "password") {
      cookie.save("auth", true);
      this.props.history.push("/search")
    } else {
      window.confirm('Invalid Credentials')
    }
  }

  render() {

    const { username, password } = this;
    return (

      <div className="loginForm">
        <form>
          <fieldset>
            <div className="form-group">
              <label >Username</label>
              <input type="text" id="username" placeholder="Username" onChange={this.handleChange} value={username} />
              <label >Password</label>
              <input type="password" id="password" placeholder="Password" onChange={this.handleChange} value={password} />
              <input id="submit" type="button" value="Login" onClick={this.handleSubmit} />
            </div>
          </fieldset>
        </form>
      </div>

    )
  }
}

export default Login;
