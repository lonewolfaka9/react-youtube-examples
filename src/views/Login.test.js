import React from 'react';
import Login from './Login';
import { configure, shallow } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
 
const adapter = ReactSixteenAdapter;
configure({ adapter: new adapter.default() });

describe('Login', () => {
    const historyMock = { push: jest.fn() }; 
    window.confirm = jest.fn(() => true) // always click 'yes'
    const login = shallow(<Login history={historyMock} />);
    it('Login Page render', () => {
        expect(login).toMatchSnapshot();
    });

    
  
});