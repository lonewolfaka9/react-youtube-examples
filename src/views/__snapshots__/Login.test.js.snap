// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login Page render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    history={
      Object {
        "push": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "maxWidth": 500,
            "minWidth": 300,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <form>
          <fieldset>
            <legend
              style={
                Object {
                  "fontSize": 20,
                  "margin": 10,
                }
              }
            >
              Login here
            </legend>
            <label
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
            >
              Username
            </label>
            <br />
            <input
              id="username"
              onChange={[Function]}
              placeholder="Username"
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="text"
              value=""
            />
            <label
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
            >
              Password
            </label>
            <input
              id="password"
              onChange={[Function]}
              placeholder="Password"
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="password"
              value=""
            />
            <input
              id="submit"
              onClick={[Function]}
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="button"
              value="Login"
            />
          </fieldset>
        </form>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <fieldset>
            <legend
              style={
                Object {
                  "fontSize": 20,
                  "margin": 10,
                }
              }
            >
              Login here
            </legend>
            <label
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
            >
              Username
            </label>
            <br />
            <input
              id="username"
              onChange={[Function]}
              placeholder="Username"
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="text"
              value=""
            />
            <label
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
            >
              Password
            </label>
            <input
              id="password"
              onChange={[Function]}
              placeholder="Password"
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="password"
              value=""
            />
            <input
              id="submit"
              onClick={[Function]}
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="button"
              value="Login"
            />
          </fieldset>
        </form>,
        "style": Object {
          "bottom": 0,
          "left": 0,
          "margin": "auto",
          "maxWidth": 500,
          "minWidth": 300,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <fieldset>
            <legend
              style={
                Object {
                  "fontSize": 20,
                  "margin": 10,
                }
              }
            >
              Login here
            </legend>
            <label
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
            >
              Username
            </label>
            <br />
            <input
              id="username"
              onChange={[Function]}
              placeholder="Username"
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="text"
              value=""
            />
            <label
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
            >
              Password
            </label>
            <input
              id="password"
              onChange={[Function]}
              placeholder="Password"
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="password"
              value=""
            />
            <input
              id="submit"
              onClick={[Function]}
              style={
                Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                }
              }
              type="button"
              value="Login"
            />
          </fieldset>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <legend
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 10,
                  }
                }
              >
                Login here
              </legend>,
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Username
              </label>,
              <br />,
              <input
                id="username"
                onChange={[Function]}
                placeholder="Username"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="text"
                value=""
              />,
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Password
              </label>,
              <input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="password"
                value=""
              />,
              <input
                id="submit"
                onClick={[Function]}
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="button"
                value="Login"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login here",
                "style": Object {
                  "fontSize": 20,
                  "margin": 10,
                },
              },
              "ref": null,
              "rendered": "Login here",
              "type": "legend",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Username",
                "style": Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                },
              },
              "ref": null,
              "rendered": "Username",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "username",
                "onChange": [Function],
                "placeholder": "Username",
                "style": Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                },
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password",
                "style": Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                },
              },
              "ref": null,
              "rendered": "Password",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "style": Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                },
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "submit",
                "onClick": [Function],
                "style": Object {
                  "display": "table-cell",
                  "fontSize": 20,
                  "margin": 10,
                  "width": "90%",
                },
                "type": "button",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "fieldset",
        },
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "margin": "auto",
              "maxWidth": 500,
              "minWidth": 300,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <form>
            <fieldset>
              <legend
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 10,
                  }
                }
              >
                Login here
              </legend>
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Username
              </label>
              <br />
              <input
                id="username"
                onChange={[Function]}
                placeholder="Username"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="text"
                value=""
              />
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Password
              </label>
              <input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="password"
                value=""
              />
              <input
                id="submit"
                onClick={[Function]}
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="button"
                value="Login"
              />
            </fieldset>
          </form>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form>
            <fieldset>
              <legend
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 10,
                  }
                }
              >
                Login here
              </legend>
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Username
              </label>
              <br />
              <input
                id="username"
                onChange={[Function]}
                placeholder="Username"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="text"
                value=""
              />
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Password
              </label>
              <input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="password"
                value=""
              />
              <input
                id="submit"
                onClick={[Function]}
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="button"
                value="Login"
              />
            </fieldset>
          </form>,
          "style": Object {
            "bottom": 0,
            "left": 0,
            "margin": "auto",
            "maxWidth": 500,
            "minWidth": 300,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <fieldset>
              <legend
                style={
                  Object {
                    "fontSize": 20,
                    "margin": 10,
                  }
                }
              >
                Login here
              </legend>
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Username
              </label>
              <br />
              <input
                id="username"
                onChange={[Function]}
                placeholder="Username"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="text"
                value=""
              />
              <label
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
              >
                Password
              </label>
              <input
                id="password"
                onChange={[Function]}
                placeholder="Password"
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="password"
                value=""
              />
              <input
                id="submit"
                onClick={[Function]}
                style={
                  Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  }
                }
                type="button"
                value="Login"
              />
            </fieldset>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <legend
                  style={
                    Object {
                      "fontSize": 20,
                      "margin": 10,
                    }
                  }
                >
                  Login here
                </legend>,
                <label
                  style={
                    Object {
                      "display": "table-cell",
                      "fontSize": 20,
                      "margin": 10,
                      "width": "90%",
                    }
                  }
                >
                  Username
                </label>,
                <br />,
                <input
                  id="username"
                  onChange={[Function]}
                  placeholder="Username"
                  style={
                    Object {
                      "display": "table-cell",
                      "fontSize": 20,
                      "margin": 10,
                      "width": "90%",
                    }
                  }
                  type="text"
                  value=""
                />,
                <label
                  style={
                    Object {
                      "display": "table-cell",
                      "fontSize": 20,
                      "margin": 10,
                      "width": "90%",
                    }
                  }
                >
                  Password
                </label>,
                <input
                  id="password"
                  onChange={[Function]}
                  placeholder="Password"
                  style={
                    Object {
                      "display": "table-cell",
                      "fontSize": 20,
                      "margin": 10,
                      "width": "90%",
                    }
                  }
                  type="password"
                  value=""
                />,
                <input
                  id="submit"
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "table-cell",
                      "fontSize": 20,
                      "margin": 10,
                      "width": "90%",
                    }
                  }
                  type="button"
                  value="Login"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login here",
                  "style": Object {
                    "fontSize": 20,
                    "margin": 10,
                  },
                },
                "ref": null,
                "rendered": "Login here",
                "type": "legend",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Username",
                  "style": Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  },
                },
                "ref": null,
                "rendered": "Username",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "username",
                  "onChange": [Function],
                  "placeholder": "Username",
                  "style": Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  },
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password",
                  "style": Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  },
                },
                "ref": null,
                "rendered": "Password",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "style": Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  },
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "submit",
                  "onClick": [Function],
                  "style": Object {
                    "display": "table-cell",
                    "fontSize": 20,
                    "margin": 10,
                    "width": "90%",
                  },
                  "type": "button",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "fieldset",
          },
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
